internal/backend/sessions/sessionworkflows/workflow.go

```diff
@@ -478,6 +482,21 @@ func (w *sessionWorkflow) run(wctx workflow.Context) (prints []string, err error
 			kittehs.ContainedIn(argNames...),
 		)
 
+		// If session was triggered by a connection, pass the connection to the entrypoint.
+		var trigger struct{ ConnectionId string }
+		if err := w.data.Session.Inputs()["trigger"].UnwrapInto(&trigger); err == nil && trigger.ConnectionId != "" {
+			if _, conn := kittehs.FindFirst(w.data.Connections, func(c sdktypes.Connection) bool {
```

Comment:
I find this style of coding, where the `; conn.IsValid` a couple of line below, confusing.
I prefer

```go
_, conn = kittehs.FindFirst(w.data.Connections, func(c sdktypes.Connection) bool {
    return c.ID().String() == trigger.ConnectionId
})
if !conn.IsValid() {
...
```